### ISSUE TITLE
Adaptive with the normalization of [-1, 1]

### ISSUE THREAD

**Issue by cxmscb** [fra31/auto-attack#16]
Hi, I found the normalization used for images is [0, 1]. If the normalization of images is [-1, 1], how could I revise the attack codes?

**Comment by fra31** [comment]
Hi,

I think the easiest solution (assuming PyTorch models) is to use images in [0, 1] and create a wrapper for the models like
```python
class NewModel():
    def __init__(self, model):
        self.model = model

    def __call__(self, x):
        z = x * 2. - 1. # z in [-1, 1] if x in [0, 1]
        return self.model(z)
```
which takes input in [0, 1], but rescale it to [-1, 1] before inference. Please refer also to https://github.com/fra31/auto-attack/issues/13 for a similar case.

For TensorFlow models, I guess one could do something similar in the model definition, i.e. adding `z = x * 2. - 1.` on the input to rescale it to [-1, 1].

Let me know if this helps!

**Comment by cxmscb** [comment]
Thanks!
